### PR TITLE
Medical Blood - Settings improvements and cleanup

### PR DESCRIPTION
**Pull request by mharis001**
**When merged this pull request will:**
- Make the `enabledFor` setting able to be changed mid-mission
- Add settings for medical blood (max objects, lifetime) similar to litter
- General cleanup of the component

### CODE REVIEW

File: addons/medical_blood/functions/fnc_init.sqf [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    [QEGVAR(medical,woundReceived), GVAR(woundReceivedEH)] call CBA_fnc_removeEventHandler;
+    GVAR(woundReceivedEH) = nil;
+};
+
+// Don't need to set up anything if blood is disabled or players only on a non-player machine
+if (_mode == BLOOD_DISABLED || {_mode == BLOOD_ONLY_PLAYERS && {!hasInterface}}) exitWith {
+    TRACE_1("Mode does not require any setup",_mode);
+};
+
+private _listCode = if (_mode == 1) then {
```
Comment (PabstMirror) [inline addons/medical_blood/functions/fnc_init.sqf]:
```suggestion
private _listCode = if (_mode == BLOOD_ONLY_PLAYERS ) then {
```

File: addons/medical_blood/functions/fnc_createBlood.sqf [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
+TRACE_2("Creating blood",_type,_position);
 
 private _model = GVAR(models) getVariable _type;
 
-private _object = createSimpleObject [_model, [0,0,0]];
-_object setDir random 360;
-_object setPos _pos;
+private _bloodDrop = createSimpleObject [_model, [0, 0, 0]];
+_bloodDrop setDir random 360;
+_bloodDrop setPos _position;
```
Comment (dedmen) [inline addons/medical_blood/functions/fnc_createBlood.sqf]: Do we need a seperate setPos? It creates a additonal network message as blooddrops are global.
Speaking of which, couldn't we make blooddrops local only with the new createSimpleObject local syntax?

Comment (SilentSpike) [reply addons/medical_blood/functions/fnc_createBlood.sqf]: I think we want them to be global, but agree about whether `setPos` is necessary

Comment (PabstMirror) [reply addons/medical_blood/functions/fnc_createBlood.sqf]: Surprising results (unless I made a mistake)

`createVehicle` @ [0,0,0] - 492 ms
```
private _start = diag_tickTime; 
for "_i" from 0 to 30000 step 3 do { 
    private _pos = (getPos player) vectorAdd [0,_i,0]; 
    private _z = "Land_LampStreet_small_F" createVehicle [0,0,0]; 
    _z setPos _pos; 
}; 
private _end = diag_tickTime; 
x3 = _end - _start;
```

`createVehicle` @ pos - 4798 ms (and non-precise placement)
```
private _start = diag_tickTime; 
for "_i" from 0 to 30000 step 3 do { 
    private _pos = (getPos player) vectorAdd [0,_i,0]; 
    private _z = "Land_LampStreet_small_F" createVehicle _pos; 
}; 
private _end = diag_tickTime; 
x3 = _end - _start;
```

`createSimpleObject` @ [0,0,0] - 263 ms
```
private _start = diag_tickTime; 
for "_i" from 0 to 30000 step 3 do { 
    private _pos = AGLtoASL ((getPos player) vectorAdd [0,_i,0]); 
    private _z = createSimpleObject ["Land_LampStreet_small_F", [0,0,0]]; 
    _z setPosASL _pos;
}; 
private _end = diag_tickTime; 
x3 = _end - _start;
```

`createSimpleObject` @ pos - 2641 ms
```
private _start = diag_tickTime; 
for "_i" from 0 to 30000 step 3 do { 
    private _pos = AGLtoASL ((getPos player) vectorAdd [0,_i,0]); 
    private _z = createSimpleObject ["Land_LampStreet_small_F", _pos]; 
}; 
private _end = diag_tickTime; 
x3 = _end - _start;
```

Comment (PabstMirror) [reply addons/medical_blood/functions/fnc_createBlood.sqf]: network traffic vs sqf speed
(unless the game can serialize the create and setPos together?)
we can continue to think about this after merge

**Pull request merged**